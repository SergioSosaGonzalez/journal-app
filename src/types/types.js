export const types = {
  login: '[Auth] login',
  logout: '[Auth] logout',

  uiSetError: '[UI] set error',
  uiRemoveError: '[UI] remove error',

  uiStartLoading: '[UI] Start loading',
  uiFinishLoading: '[UI] Finish loading',

  notesAddNew: '[Notes] New Note',
  notesActive: '[Notes] Set Active Note',
  notes: '[Notes] Load Note',
  notesUpdated: '[Notes] Update Note',
  notesLoad: '[Notes] Load Note',
  notesFileUrl: '[Notes] Updated image url',
  notesDeleted: '[Notes] Delete note',
  notesLogoutCleaning: '[Notes] Logout cleaning',
};
