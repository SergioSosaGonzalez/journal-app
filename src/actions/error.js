import { types } from '../types/types';

export const setError = (text) => ({ type: types.uiSetError, payload: text });

export const removeError = () => ({
  type: types.uiRemoveError,
});
