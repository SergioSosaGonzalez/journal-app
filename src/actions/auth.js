import { types } from '../types/types';
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import Swal from 'sweetalert2';
import { notesLogout } from './notes';

export const startLoginWithEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading, payload: {} });
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch({ type: types.uiFinishLoading, payload: {} });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
        dispatch({ type: types.uiFinishLoading, payload: {} });
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    console.log('hola mudno');
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const logout = () => {
  return { type: types.logout };
};

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogout());
  };
};
