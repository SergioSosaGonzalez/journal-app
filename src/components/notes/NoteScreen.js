import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { activeNote, startDeleting } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NoteAppbar } from './NoteAppbar';

const NoteScreen = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const [formValues, handleInputChange, reset] = useForm(note);
  const { body, title, url } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    dispatch(activeNote(note.id, { ...formValues }));
  }, [formValues, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleting(note.id));
  };
  return (
    <div className="notes__main-content">
      <NoteAppbar />
      <div className="notes__content">
        <form>
          <input
            type="text"
            placeholder="Some awesome title"
            className="notes__title-input"
            value={title}
            name="title"
            onChange={handleInputChange}
          />
          <textarea
            placeholder="What happen today"
            className="notes__text-area"
            value={body}
            name="body"
            onChange={handleInputChange}></textarea>
          <div className="notes__image">
            {note.url && <img src={note.url} alt="img" />}
          </div>
        </form>
      </div>

      <button className="btn btn-danger" onClick={handleDelete}>
        Delete
      </button>
    </div>
  );
};
export default NoteScreen;
