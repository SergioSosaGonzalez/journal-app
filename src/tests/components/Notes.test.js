import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../types/types';
import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import NoteScreen from '../../components/notes/NoteScreen';
import { activeNote } from '../../actions/notes';

jest.mock('../../actions/notes', () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: 1234,
      title: 'Hola',
      body: 'mundo',
      date: 0,
    },
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

describe('Pruebas de NotesScreen', () => {
  const wrapper = mount(
    <Provider store={store}>
      <NoteScreen />
    </Provider>
  );
  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de disparar el activeNote', () => {
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'hola de nuevo prro',
      },
    });
    expect(activeNote).toHaveBeenCalled();
    expect(activeNote).toHaveBeenLastCalledWith(1234, {
      body: 'mundo',
      title: 'hola de nuevo prro',
      id: 1234,
      date: 0,
    });
  });
});
