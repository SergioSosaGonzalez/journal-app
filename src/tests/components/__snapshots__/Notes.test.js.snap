// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de NotesScreen Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
       28 de agosto del 2020
    </span>
    <input
      id="fileSelector"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
         
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
         
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <form>
      <input
        className="notes__title-input"
        name="title"
        onChange={[Function]}
        placeholder="Some awesome title"
        type="text"
        value="Hola"
      />
      <textarea
        className="notes__text-area"
        name="body"
        onChange={[Function]}
        placeholder="What happen today"
        value="mundo"
      />
      <div
        className="notes__image"
      />
    </form>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
